### PR TITLE
feat(provider): wire middleware pipeline, delete FallbackChainExecutor

### PR DESCRIPTION
**Pull request by CybotTM**
## Summary

**Final step of the ADR-026 follow-up chain.** Switches `LlmServiceManager`'s DB-backed configuration paths over to `MiddlewarePipeline` and removes the legacy `FallbackChainExecutor` + its tests. Every `ProviderMiddlewareInterface` implementation tagged `nr_llm.provider_middleware` (Fallback, Budget, Usage, Cache — all landed in prior PRs) now wraps the call automatically.

Closes the ADR.

### `LlmServiceManager`
- Constructor: drops the optional `?FallbackChainExecutor $fallbackChainExecutor = null`. Adds required `MiddlewarePipeline $pipeline`.
- Private `runWithFallback()` → `runThroughPipeline()`. Now invokes `$this->pipeline->run($context, $configuration, $terminal)` with a `ProviderCallContext::for($operation)`.
- `chatWithConfiguration()` passes `ProviderOperation::Chat`; `completeWithConfiguration()` passes `ProviderOperation::Completion`. Their terminal closures are unchanged.
- `streamChatWithConfiguration()` is deliberately **not** routed through the pipeline — streaming calls cannot be retried mid-stream (documented in `FallbackMiddleware`).

### Deleted
- `Classes/Service/FallbackChainExecutor.php`
- `Tests/Unit/Service/FallbackChainExecutorTest.php`
- `Tests/Integration/Service/FallbackChainIntegrationTest.php`

Their behavioural coverage is subsumed by `FallbackMiddlewareTest` (landed in #138) which mirrors the same retryable-exception matrix, chain-walk semantics, and edge cases — routed through `MiddlewarePipeline::run()` instead of `FallbackChainExecutor::execute()`.

### Test updates
Six test files patched to pass an empty `MiddlewarePipeline([])` as the new required constructor arg — identical behaviour to before (empty pipeline just runs the terminal). Files:
- `Tests/Unit/Service/LlmServiceManagerTest.php`
- `Tests/Unit/Service/LlmServiceManagerMutationTest.php`
- `Tests/Integration/Service/LlmServiceManagerIntegrationTest.php`
- `Tests/Unit/Specialized/Translation/LlmTranslatorTest.php`
- `Tests/E2E/ChatCompletionWorkflowTest.php`
- `Tests/E2E/EmbeddingWorkflowTest.php`

## Deliberately out of scope (separate PRs as discussed)

- **Direct methods (`chat`/`complete`/`embed`/`vision`) through the pipeline.** They take a raw provider id with no `LlmConfiguration` to thread through. Unifying that is a migration, not a refactor.
- **Switching `EmbeddingService` cache branch over to `CacheMiddleware`.** The middleware exists; consumers opt in per call when they want to.
- **Populating `beUserUid` / `cacheKey` / `plannedCost` metadata at feature-service boundaries.** Callers opt in as they do today; the plumbing just works when they do.

## Known compatibility break

Third-party DI overrides injecting `FallbackChainExecutor` into `LlmServiceManager` will break at container boot. This is the point of the refactor and the ADR documents it. If anyone hits this downstream, the migration is a one-line `Services.yaml` swap to `MiddlewarePipeline`.

## Verification

- **3136 unit tests** green on PHP 8.4
- PHPStan level 10 clean
- PHP-CS-Fixer dry-run clean
- Architecture tests green (no layer violations from the deletion)
- Diff stat: **+64 / −923** — net 859 lines deleted, reflecting the executor replacement by the already-landed middleware

## ADR-026 roadmap — complete

| # | Step | PR | State |
|---|---|---|---|
| 0 | Middleware infrastructure | #137 | Merged |
| 1 | FallbackMiddleware | #138 | Merged |
| 2 | BudgetMiddleware | #139 | Merged |
| 3 | UsageMiddleware | #141 | Merged |
| 4 | CacheMiddleware | #142 | Merged |
| **5** | **Feature wiring + executor deletion** | **This PR** | **Open** |

Audit recommendation #1 — fully discharged.

## Test plan
- [ ] CI green
- [ ] Spot-check `LlmServiceManager::chatWithConfiguration()` / `completeWithConfiguration()` call sites in any downstream code that mocks or extends them
- [ ] Confirm `Services.yaml` entry for `LlmServiceManager` still wires correctly (autowiring should pick up `MiddlewarePipeline` — no manual entry needed)

### CODE REVIEW

File: Classes/Service/LlmServiceManager.php [hunk 1]
```diff
@@ -374,24 +379,30 @@ function (LlmConfiguration $config) use ($prompt): CompletionResponse {
...
+        LlmConfiguration $configuration,
+        ProviderOperation $operation,
+        callable $terminal,
+    ): mixed {
+        return $this->pipeline->run(
+            ProviderCallContext::for($operation),
+            $configuration,
+            $terminal,
+        );
     }
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Service/LlmServiceManager.php:406]: Addressed in follow-up PR #152 — LlmServiceManagerTest now has  (plus four siblings for complete/embed/vision/chatWithTools) using a  spy that asserts operation, identifier, fallback-chain shape, uid, and context metadata. See PR #150 (FU-5 direct-method wiring) for the initial tests and PR #152 for the metadata assertions.

File: Tests/Unit/Service/LlmServiceManagerMutationTest.php [hunk 2]
```diff
@@ -37,7 +38,7 @@ private function createManager(array $config = []): LlmServiceManager
         $loggerStub = self::createStub(LoggerInterface::class);
         $adapterRegistryStub = self::createStub(ProviderAdapterRegistry::class);
 
-        return new LlmServiceManager($extensionConfigStub, $loggerStub, $adapterRegistryStub);
+        return new LlmServiceManager($extensionConfigStub, $loggerStub, $adapterRegistryStub, new MiddlewarePipeline([]));
```
[the remark this answers is not in the document]
Comment (CybotTM) [reply Tests/Unit/Service/LlmServiceManagerMutationTest.php:41]: Addressed in follow-up PR #152 — added . 15 sites across LlmServiceManagerTest, LlmServiceManagerMutationTest, and LlmTranslatorTest now go through the helper. E2E and integration tests keep explicit construction (they don't extend AbstractUnitTestCase).

File: Classes/Service/LlmServiceManager.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (CybotTM) [reply Classes/Service/LlmServiceManager.php:406]: (Correction to stripped reply above — the missing names: the new test is `directChatRoutesThroughMiddlewarePipeline` in `LlmServiceManagerTest`, using a `RecordingMiddleware` spy.)
